lib/constructor.js

```diff
@@ -357,6 +357,7 @@ const Sharp = function (input, options) {
     }
   };
   this.options.input = this._createInputDescriptor(input, options, { allowStream: true });
+  this.options.useExifOrientation = !!this.options.input.autoOrient;
```

Comment:
Not sure if you'd rather do this or rename useExifOrientation. The main part of the pipeline is looking for `useExifOrientation` to do auto orientation, while the composite part of the pipeline looks for `autoOrient`.

Reply:
Update: I went ahead and replaced `options.useExifOrientation` with `options.input.autoOrient` everywhere, and let that cascade into the C++ stuff as well.